File: source/source/cs12/CustomControl/HelloCustom.cs
using System;
using System.Web.UI;

namespace CustomControl
{
	/// <summary>
	/// Summary description for HelloCustom.
	/// </summary>
	public class HelloCustom : Control
	{
		public HelloCustom() 
		{
			//
			// TODO: Add constructor logic here
			//
		}

		// Text 프로퍼티
		private string m_Text;
		public string Text
		{
			get
			{
				return m_Text;
			}
			set
			{
				m_Text = value;
			}
		}

		protected override void Render(System.Web.UI.HtmlTextWriter writer)
		{
			writer.Write(m_Text);
		}
	}


}

File: source/source/vb12/CustomControl/HelloCustom.vb
Imports System.Web.UI

Public Class HelloCustom
    Inherits Control

    ' Text 프로퍼티 
    Private m_Text As String = "Hello, Custom Control!"
    Public Property Text() As String
        Get
            Return m_Text
        End Get
        Set(ByVal Value As String)
            m_Text = Value
        End Set
    End Property


    Protected Overrides Sub Render(ByVal writer As System.Web.UI.HtmlTextWriter)
        writer.Write(m_Text)
    End Sub

End Class